src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -298,6 +371,34 @@ private static void ChangeColor(bool foreground, ConsoleColor color)
         /// <summary>Cache of the format strings for foreground/background and ConsoleColor.</summary>
         private static readonly string[,] s_fgbgAndColorStrings = new string[2, 16]; // 2 == fg vs bg, 16 == ConsoleColor values
 
+        public static bool TryGetSpecialConsoleKey(char[] givenChars, int startIndex, int endIndex, out ConsoleKey key, out int keyLength)
+        {
+            key = (ConsoleKey)0;
+            keyLength = 0;
+            int unprocessedCharCount = endIndex - startIndex + 1;
+
+            if (unprocessedCharCount < TerminalKeyInfo.Instance.MinKeyLength)
```

Comment:
It answers my question, but also confirms my concern :smile:  I realize it's very unlikely that we'll hit such a situation, but if we do we'll end up potentially interpreting the data incorrectly.  I'm wondering if, when we have more than 0 but fewer than MaxKeyLength characters, we should poll to see if there's any more data available from stdin, and if there is, populate the buffer with it?